CMakeLists.txt

```diff
@@ -131,10 +131,21 @@ include("cmake/host_compiler.cmake")
 include("cmake/configuring_primitive_list.cmake")
 
 if(UNIX OR MINGW)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
-    # Let SYCL to choose the C++ standard it needs.
-    if(NOT DNNL_WITH_SYCL)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+    if(CMAKE_VERSION VERSION_LESS "3.1.0")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+        # Let SYCL to choose the C++ standard it needs.
+        if(NOT DNNL_WITH_SYCL)
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+        endif()
+    else()
+        # Use CMake 3.1 functionality and allow overriding to
+        # higher values
+        if(NOT DEFINED CMAKE_C_STANDARD AND CMAKE_C_STANDARD_DEFAULT EQUAL 89)
+            set(CMAKE_C_STANDARD 99)
+        endif()
+        if(NOT DEFINED CMAKE_CXX_STANDARD AND CMAKE_CXX_STANDARD_DEFAULT EQUAL 98)
```

Comment:
I was actually thinking about a little different logic here. I think we should always use C++11 and C99 standards unless `CMAKE_C_STANDARD` and `CMAKE_CXX_STANDARD` override it.
But those users who want to build oneDNN with newer standards should be able to do that by specifying them via the CMake variables.

The are two reasons for that:
1. We don't want to change user visible behavior, it's expected that oneDNN uses C++11 by default
2. We have an extensive validation with wide variety of compilers therefore we want to fix the standard version to make sure the default configuration works as expected

Reply:
Ok, I see. Let me try that.